Mathlib/CategoryTheory/FiberedCategory/Fiber.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Calle Sönne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Calle Sönne, Paul Lezeau
+-/
+
+import Mathlib.CategoryTheory.FiberedCategory.HomLift
+import Mathlib.CategoryTheory.Functor.Const
+
+/-!
+
+# Fibers of a functors
+
+In this file we define, for a functor `p : 𝒳 ⥤ 𝒴`, the fiber categories `Fiber p S` for every
+`S : 𝒮` as follows
+- An object in `Fiber p S` is a pair `(a, ha)` where `a : 𝒳` and `ha : p.obj a = S`.
+- A morphism in `Fiber p S` is a morphism `φ : a ⟶ b` in 𝒳 such that `p.map φ = 𝟙 S`.
+
+For any category `C` equipped with a functor `F : C ⥤ 𝒳` such that `F ⋙ p` is constant at `S`,
+we define a functor `InducedFunctor : C ⥤ Fiber p S` that `F` factors through.
+-/
+
+universe v₁ u₁ v₂ u₂ v₃ u₃
+
+open CategoryTheory Functor Category IsHomLift
+
+variable {𝒮 : Type u₁} {𝒳 : Type u₂} [Category.{v₁} 𝒮] [Category.{v₂} 𝒳]
+
+/-- `Fiber p S` is the type of elements of `𝒳` mapping to `S` via `p`.  -/
+def Fiber (p : 𝒳 ⥤ 𝒮) (S : 𝒮) := {a : 𝒳 // p.obj a = S}
+
+namespace Fiber
+
+/-- `Hom a b` are the morphisms of `Fiber p S`, defined as those lying over `𝟙 S` in the base. -/
+def Hom {p : 𝒳 ⥤ 𝒮} {S : 𝒮} (a b : Fiber p S) := {φ : a.1 ⟶ b.1 // IsHomLift p (𝟙 S) φ}
```

Comment:
This spacing is more standard in mathlib:
```suggestion
def Hom {p : 𝒳 ⥤ 𝒮} {S : 𝒮} (a b : Fiber p S) := { φ : a.1 ⟶ b.1 // IsHomLift p (𝟙 S) φ }
```

Reply:
This definition is probably unnecessary, because you define it directly in the fiberCategory instance.

Reply:
I have decided to use the terminology `Hom` also in the `fiberCategory` instance. The point of this definition was so that I could add the instance below it (which simplifies the proof that composition of hom's is a hom)